Flaky test "kill normal client"

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-kvrocks/issues) and found no similar issues.


### Motivation

```
--- FAIL: TestIntrospection (3.80s)
    --- FAIL: TestIntrospection/Kill_normal_client (0.00s)
        introspection_test.go:120: 
            	Error Trace:	/home/runner/work/incubator-kvrocks/incubator-kvrocks/tests/gocase/unit/introspection/introspection_test.go:120
            	Error:      	Not equal: 
            	            	expected: int(1)
            	            	actual  : int64(2)
            	Test:       	TestIntrospection/Kill_normal_client
```

https://github.com/apache/incubator-kvrocks/actions/runs/3286622831/jobs/5414949088

### Solution

Perhaps we should tolerate 1 or 2.

@vmihailenco is this behavior related to go-redis auto reconnecting?

cc @git-hulk 

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

--- a/src/server/worker.cc
+++ b/src/server/worker.cc
@@ -430,7 +430,7 @@ void Worker::KillClient(Redis::Connection *self, uint64_t id, std::string addr, 
   for (const auto &iter : conns_) {
     Redis::Connection *conn = iter.second;
     if (skipme && self == conn) continue;
-    // no need to kill again the client if flags as kCloseAfterReply
+    // no need to kill the client again if flags as kCloseAfterReply
     if (conn->IsFlagEnabled(Redis::Connection::kCloseAfterReply)) {
       continue;
     }

--- a/src/server/worker.cc
+++ b/src/server/worker.cc
@@ -430,6 +430,10 @@ void Worker::KillClient(Redis::Connection *self, uint64_t id, std::string addr, 
   for (const auto &iter : conns_) {
     Redis::Connection *conn = iter.second;
     if (skipme && self == conn) continue;
+    // no need to kill again the client if flags as kCloseAfterReply
+    if (conn->IsFlagEnabled(Redis::Connection::kCloseAfterReply)) {
+      continue;
+    }
     if ((type & conn->GetClientType()) || (!addr.empty() && conn->GetAddr() == addr) ||
         (id != 0 && conn->GetID() == id)) {
       conn->EnableFlag(Redis::Connection::kCloseAfterReply);

